types/autocomplete.js.flow

```diff
@@ -1,21 +1,22 @@
 /* @flow */
 
-// One of text or snippet is required.
-// TODO(hansonw): use a union + intersection type
```

Comment:
This todo is suggesting a good change to the one require problem. Something like

```
type AutocompleteSuggestionText = {| text: string |} & _AutocompleteSuggestionPartial
type AutocompleteSuggestionSnippet = {| snippet: string |} & _AutocompleteSuggestionPartial
type _AutocompleteSuggestionPartial = {|
  replacementPrefix: string,
  replacedText: string,
  iconHTML?: string,
  type?: string,
  description?: string,
  descriptionMoreURL?: string,
  displayText?: string,
  leftLabel?: string,
  leftLabelHTML?: string,
  rightLabel?: string,
  rightLabelHTML?: string,
  className?: string
|}
type atom$AutocompleteSuggestion = AutocompleteSuggestionText  | AutocompleteSuggestionSnippet

Reply:
Do you this is this worth to do ?

Reply:
I already did it, and I think it is better than setting test required and not snippet.